sound/soc/sof/intel/shim.h

```diff
@@ -185,6 +185,8 @@ struct sof_intel_dsp_desc {
 	u32 d0i3_offset;
 	u32 quirks;
 	enum sof_intel_hw_ip_version hw_ip_version;
+	u32 interface_mask;		/* supported: BIT(SOF_DAI_INTEL_XXX) set */
+	u32 interface_mask_dsp_only;	/* supported only by DSP: BIT(SOF_DAI_INTEL_XXX) set */
```

Comment:
If we add the mask like this:
```c
u32 interface_mask[2];
```
Optionally add two defines:
```c
#define SOF_HDA_INTERFACES_WITH_DSP	0
#define SOF_HDA_INTERFACES_WITHOUT_DSP	1
```
and in tgl.c for example:
```c
	.interface_mask = {
		[SOF_HDA_INTERFACES_WITH_DSP] = BIT(SOF_DAI_INTEL_SSP) |
						BIT(SOF_DAI_INTEL_DMIC) |
						BIT(SOF_DAI_INTEL_HDA) |
						BIT(SOF_DAI_INTEL_ALH),
		[SOF_HDA_INTERFACES_WITHOUT_DSP] = BIT(SOF_DAI_INTEL_HDA),
	},
```
then in code we could just check for example the SDW availability:
```
	const struct sof_intel_dsp_desc *chip = get_chip_info(pdata);

	if ((chip->interface_mask[sdev->dspless_mode_selected] & BIT(SOF_DAI_INTEL_ALH))) {
		// SDW is supported in this setup
	} else {
		// SDW is not supported in this setup
	}
```

Reply:
@plbossart, or what about this:
https://github.com/thesofproject/linux/pull/3962/commits/852ee40a7d82c9049202d1d242cb1ee7c9f1853c
`hda_get_interface_mask()` in hda.c

Reply:
yes, that's what I was suggesting as an alternative, and in that function we use the architecture version to find out what the supported interfaces are. No need to hard code the same thing for multiple chips that are all the same between SKL and MTL.